src/monobase/refresh_files.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python3
+
+import argparse
+import hashlib
+import json
+import os
+import re
+import shutil
+import subprocess
+import sys
+import urllib
+import urllib.request
+from http import HTTPStatus
+
+MONOBASE_PREFIX = os.environ.get('MONOBASE_PREFIX', '/srv/r8/monobase')
+PGET_BIN = os.environ.get('PGET_BIN', os.path.join(MONOBASE_PREFIX, 'bin/pget-bin'))
+KNOWN_WEIGHTS_DIR = os.environ.get('KNOWN_WEIGHTS_DIR', '')
+
+parser = argparse.ArgumentParser('refresh_files')
+parser.add_argument('-q', '--query-url', type=str)
+parser.add_argument('-u', '--upstream-url', type=str)
+parser.add_argument('-a', '--auth-token', type=str)
+
+
+def find_pget_exe() -> str:
+    # PGET_BIN is executable
+    if os.path.isfile(PGET_BIN) and os.access(PGET_BIN, os.X_OK):
+        return PGET_BIN
+    # Look for real executable in PATH
+    for p in os.environ['PATH'].split(os.pathsep):
+        f = shutil.which('pget', path=p)
+        if f is not None and f != __file__:
+            return f
+    print('Cannot find pget executable', file=sys.stderr)
+    sys.exit(1)
+
+
+def main(url, upstream, auth_token) -> None:
+    if not url:
+        print('Misconfigured files URL, exiting')
+        sys.exit(1)
+
+    try:
+        req = urllib.request.Request(
+            url, method='GET', headers={'X-Honeycomb-Team': auth_token}
+        )
+        resp = urllib.request.urlopen(req)
+    except urllib.error.HTTPError as e:
+        print(f'Failed request to {url}: {e.read().decode()}; exiting')
+        sys.exit(1)
+
+    if resp.status != HTTPStatus.OK:
+        print(f'Failed request to {url}: {resp.read().decode()}; exiting')
+        sys.exit(1)
+
+    # The expectation for response from the file-stats-url is that
+    # it is a GET request that returns a list of files, like
+    # {
+    #   "data": {
+    #       "results": [
+    #           {
+    #               "cache.request.resolved_url": "URL1",
+    #           },
+    #           {
+    #               "cache.request.resolved_url": "URL2",
+    #           },
+    #           ...,
+    #       ],
+    #   }
+    # }
+    body = json.loads(resp.read().decode())
+    results = body.get('data', {}).get('results', [])
+    if len(results) == 0:
+        print(f'Malformatted response: {body}; exiting')
+        sys.exit(1)
+
+    p = find_pget_exe()
+    file_set = set([])
+    # pget each of the files into the new directory
+    for result in results:
```

Comment:
With `--max-arg`, we can do `total += file.size()` again for every new file downloaded, and break early once we hit the max size. Assuming we have `LIMIT 1000` and average file size > 1GB, we should usually hit it early.